scripts/sanity_chk/sanitylib.py

```diff
@@ -2708,97 +2731,77 @@ def apply_filters(self, **kwargs):
                                 instance.run = True
 
                 if not force_platform and plat.name in exclude_platform:
-                    discards[instance] = "Platform is excluded on command line."
-                    continue
+                    discards[instance] = discards.get(instance, "Platform is excluded on command line")
```

Comment:
why do you need to do this?

Reply:
I updated all the cases where an instance is added to discards, just for consistency. I guess this is a more specific case involving user activity, which shouldn't matter for CI runs.  Do you see issues that can be caused by this? I don't mind changing this back.

Reply:
this reads like we are filtering the same instance twice and setting the reason to something that was previously set, IMO it is confusing. instances should be filtered only once, if this is not the case, then that is a bug.

Reply:
I'm not sure if I follow. Do you mean it's confusing because this line is the first place where discarding can occur, hence it is better to use `discards[instance] = "Platform is excluded on command line."` instead of `discards[instance] = discards.get(instance, "Platform is excluded on command line")`?  If so I see the logic and can change this.
 I saw there are multiple reasons why an instance can be discarded. E.g. tests for sensors can have "Not in testcase platform whitelist" or "No hardware support". I thought you suggested to only save the first occurring reason and this is how I've done it. Now, if there was a reason before it stays and nothing changes. If not (it is the first occurrence), the new reason is given.  I think having all reasons handled in the same way will require less effort in the future if someone decides to change the ordering in the code or add sth new before the first discard occurrence.
I would even opt for concatenating reasons, so all of them are  seen and if someone wants to fix his test/platform so it fulfills all of the conditions he/she wouldn't have to figure them out one by one.

Reply:
>  Now, if there was a reason before it stays and nothing changes.

why would there be previous reason? Once the discard reason for an instance is set, we continue to the next instance. We just take the first filter match, although some other filters could also match, but we never recorded multiple reasons, only the first hit.

> I would even opt for concatenating reasons

But we never filter the same instance twice and going through all the filters is just going to be time consuming.

Reply:
> why would there be previous reason? Once the discard reason for an instance is set, we continue to the next instance. 

This is IMO where the change is needed, which I propose with this PR. If we continue to the next instance we skip the part where 'skipped' instance is added to the instance list:
```                 
# if nothing stopped us until now, it means this configuration
# needs to be added.
instance_list.append(instance)
```
and later on not added to the TestSuite
```
self.add_instances(instance_list)
```
If these instances are not added than we cannot have results for them.
> But we never filter the same instance twice and going through all the filters is just going to be time consuming.

I am not sure if skipping `continue`s would introduce any significant execution time. I checked this with our 'on-target' setup and got ~30000 extra skipped test cases in the report but I didn't observe any difference in the CI execution time.